pkg/apis/dmi/v1/api.proto

```diff
@@ -0,0 +1,349 @@
+/*
+Copyright 2022 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// To regenerate api.pb.go run hack/generate-dmi.sh
+syntax = "proto3";
+
+//option go_package = "path;name";
+option go_package="./;v1";
+package v1;
+
+import "google/protobuf/any.proto";
+
+// DeviceManagerService defines the public APIS for remote device management.
+// The server is implemented by the module of device manager in edgecore
+// and the client is implemented by the device mapper for upstreaming.
+service DeviceManagerService {
+    // MapperRegister register the information of the mapper to device manager
+    rpc MapperRegister(MapperRegisterRequest) returns (MapperRegisterResponse) {}
+    // ReportDeviceStatus report the status of devices to device manager
+    rpc ReportDeviceStatus(ReportDeviceStatusRequest) returns (ReportDeviceStatusResponse) {}
+}
+
+// DeviceMapperService defines the public APIS for remote device management.
+// The server is implemented by the device mapper
+// and the client is implemented by the module of device manager in edgecore for downstreaming.
+service DeviceMapperService {
+    // CreateDevice create a device to the device mapper
+    rpc CreateDevice(CreateDeviceRequest) returns (CreateDeviceResponse) {}
+    // RemoveDevice remove a device to the device mapper
+    rpc RemoveDevice(RemoveDeviceRequest) returns (RemoveDeviceResponse) {}
+    // UpdateDevice update a device to the device mapper
+    rpc UpdateDevice(UpdateDeviceRequest) returns (UpdateDeviceResponse) {}
+    // CreateDeviceModel create a device model to the device mapper
+    rpc CreateDeviceModel(CreateDeviceModelRequest) returns (CreateDeviceModelResponse) {}
+    // RemoveDeviceModel remove a device model to the device mapper
+    rpc RemoveDeviceModel(RemoveDeviceModelRequest) returns (RemoveDeviceModelResponse) {}
+    // UpdateDeviceModel update a device model to the device mapper
+    rpc UpdateDeviceModel(UpdateDeviceModelRequest) returns (UpdateDeviceModelResponse) {}
```

Comment:
Could you explain why device mapper need to aware the device model info?

Reply:
the device mapper will need the information of device properties in the device model to collect datas from device and format them. under the DMI framework, the device mapper is transferred to the mapper through DMI not configmap.